### ISSUE TITLE
算子定义生成体系建设--静态图算子自动生成-第二期

### ISSUE THREAD

**Issue by heavyrain-lzy** [PaddlePaddle/Paddle#53267]
## 问题描述
> This project will be mentored by @heavyrain-lzy,@zyfncg

大家好，目前飞桨的算子已十分丰富，能够满足众多用户需求，但另一方面，繁多的算子给框架的维护和开发带来了困难。为了规范静态图算子的定义方式加快算子开发流程，飞桨建立了一套自动代码生成体系。但目前并没有将所有的算子清理完毕，这里筛选出部分简单的算子，欢迎大家一起提交清理。任务目标是清理`legacy_ops.yaml` `legacy_backward.yaml`中的OP的配置**并将原始手写的算子实现进行删除,也就是删除对应的`xxx_op.cc`和`xxx_sig.cc`文件或者文件的一部分**。前期已经开展过一次开源任务，详见[《第一期"快乐开源"框架静态图算子自动生成》](https://github.com/PaddlePaddle/Paddle/issues/51842),本次第二期的算子开发任务难度普遍增大，建议参加过第一期的同学来认领该任务，如果没有参加过第一期的开源任务的同学也想认领该工作，建议先参考第一期任务进行学习，其中相关的PR有大量详细的review意见，强烈建议学习。在第一期里列出的学习建议，这里不再重复描述。
### 注意事项
1. **学习任务开发步骤 [任务开发详细说明](https://github.com/PaddlePaddle/community/blob/master/pfcc/call-for-contributions/paddle_autogen_code_2.md)** ‼️ **极其重要** ‼️
2. **认领规则：直接回复下 issue 下方**
3. **PR 通过 CI 后，可以评论里或者 review request @heavyrain-lzy @zyfncg** ，研发会进行审核
4. 这些任务时间：PR 截止提交时间6月15日，截止合入时间6月18日
5. 这些任务有一定的难度，可以加深对框架的熟悉程度，增强代码调试能力，欢迎参与
6. [历史上的good first issue](https://github.com/PaddlePaddle/community/tree/master/pfcc#good-first-issue) 也欢迎来提 PR 解决~
7. 欢迎联系花花加入社区，和我们一起快乐开源！

<img width="200" alt="image" src="https://user-images.githubusercontent.com/39876205/213131273-361d2bb8-ea48-4f62-9603-0b0401f50337.png">

### 任务列表 （整体进度 46/46）
> 按 merge 的时间顺序，排名不分先后： @GreatV (8) @huangjiyi (10) @gouzil (6) @Liyulingyue (2) @enkilee (2) @longranger2 (2) @ccsuzzh (2) @sanbuphy (1)
> 取消 (13)

|任务序号|算子名|困难程度|认领人|相关PR
|------|-----|-------|----|----|
|1|~~transpose~~|中等| @Liyulingyue |~~#54155~~任务取消
|2|abs✅(2023/5/16)|容易| @enkilee |#53341
|3|~~add_n~~|困难| @enkilee |~~#53403~~ 暂时取消  |
|4|~~batch_norm~~|困难| @gouzil |~~#53816~~任务取消|
|5|~~sync_batch_norm_~~|困难|任务取消|
|6|bincount✅(2023/6/26)|简单| @sanbuphy | #54686 
|7|~~cast~~|中等| @huangjiyi |任务取消|
|8|~~concat~~|中等|@Liyulingyue|~~#54217~~任务取消|
|9|expand✅(2023/6/30)|容易| @ccsuzzh  | #54628
|10|exponential_✅(2023/6/20)|容易| @GreatV  | #54626
|11|gather✅(2023/6/26)|中等| @enkilee | #54084 
|12|lstsq✅(2023/5/10)|容易|@liyulingyue|#53290
|13|matmul✅(2023/6/20)|困难| @GreatV | #54338
|14|~~repeat_interleave~~|困难| @enkilee | ~~#54086~~任务取消
|15|~~repeat_interleave_with_tensor_index~~|困难| @enkilee |~~#54086~~任务取消
|16|shape✅(2023/6/2)|中等| @GreatV | #54221
|17|strided_slice✅(2023/5/30)|中等|@huangjiyi|#54098 |
|18|~~tile~~|容易| @gouzil | ~~#53317取消~~
|19|triangular_solve✅(2023/4/27)|容易| @gouzil | #53328
|20|tril✅(2023/5/23)|困难|  @gouzil | #54033
|21|triu✅(2023/5/23)|困难| @gouzil   |#54033
|22|group_norm✅(2023/5/23)|容易| @longranger2 | #53489 
|23|~~norm~~|容易| @longranger2 |~~#53491~~任务取消|
|24|prior_box✅(2023/6/15)|容易| @ccsuzzh | #54508
|25|~~split~~|困难| @longranger2 | 任务取消|
|26|~~split_with_num~~|困难|| 任务取消|
|27|einsum✅(2023/7/3)|容易| @longranger2 | #54723 
|28|fill_any✅(2023/6/14)|中等| @huangjiyi  | #54378
|29|~~fill~~ |中等| @jinyouzhi | ~~#53461~~ 任务取消|
|30|pool2d✅(2023/6/5)|困难| @gouzil | #54070
|31|pool3d✅(2023/6/5)|困难|@gouzil | #54070
|32|check_finite_and_unscale_✅(2023/6/1)|容易|@GreatV|#54145
|33|coalesce_tensor✅(2023/5/9)|中等|  @gouzil | #53570
|34|conv2d✅(2023/6/1)|困难| @huangjiyi | #54201 
|35|conv3d✅(2023/6/1)|困难| @huangjiyi | #54201 
|36|depthwise_conv2d✅(2023/6/1)|困难| @huangjiyi  | #54201 
|37|conv2d_transpose✅(2023/6/5)|困难| @huangjiyi | #54242
|38|conv3d_transpose✅(2023/6/5)|困难|@huangjiyi | #54242
|39|depthwise_conv2d_transpose✅(2023/6/5)|困难|@huangjiyi | #54242
|40|softmax✅(2023/5/16)|中等| @GreatV | #53581
|41|pad3d✅(2023/5/23)|中等| @GreatV | #53733
|42|multiclass_nms3✅(2023/6/5)|困难| @huangjiyi | #54272
|43|box_coder✅(2023/4/26)|容易| @GreatV | #53309 
|44|edit_distance✅(2023/4/26)|容易| @huangjiyi |#53297|
|45|instance_norm✅(2023/5/16)|容易| @Liyulingyue |#53340
|46|full_like✅(2023/6/19)|容易| @GreatV  |#54698

**Comment by longranger2** [comment]
认领 1，2，6，7

**Comment by longranger2** [comment]
认领23，27

**Comment by kendaji12138** [comment]
认领 9,10

**Comment by huangjiyi** [comment]
|  任务序号  | 算子名  | 困难程度 | 认领人 | 相关 PR |
|  ----           | ----  | ----  | ----  | ----  |
| 7  | cast | 中等 | @huangjiyi | #53513 (closed) |
| 44  | edit_distance | 容易 | @huangjiyi | #53297 (merged) |

**Comment by Liyulingyue** [comment]
12

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
43 | box_coder | 容易| @GreatV|#53309

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
40 | softmax | 中等 | @GreatV | #53381

**Comment by enkilee** [comment]
<html>
<body>


任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
8 | concat | 中等 | @enkilee |  https://github.com/PaddlePaddle/Paddle/pull/54056


</body>
</html>

**Comment by jinyouzhi** [comment]
|任务序号|算子名|困难程度|认领人|相关PR|
|:-:|:-:|:-:|:-:|:-:|
|29|fill|中等|@jinyouzhi|https://github.com/PaddlePaddle/Paddle/pull/53461|

**Comment by gouzil** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
33 | coalesce_tensor | 中等 | @gouzil | #53570

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
11 | <del>gather</del>| 中等 | <del>@GreatV</del>|

**Comment by Difers** [comment]
认领32

**Comment by enkilee** [comment]
<html>
<body>


任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
11 | gather | 中等 |  @enkilee | https://github.com/PaddlePaddle/Paddle/pull/54084


</body>
</html>

**Comment by enkilee** [comment]
<html>
<body>


任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
14 | repeat_interleave | 困难 |  @enkilee  |  https://github.com/PaddlePaddle/Paddle/pull/54086
15 | repeat_interleave_with_tensor_index | 困难 |  @enkilee  |  https://github.com/PaddlePaddle/Paddle/pull/54086

</body>
</html>

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
32 | check_finite_and_unscale_ | 容易 |@GreatV|#54145|

**Comment by huangjiyi** [comment]
认领 34, 35, 36

36 的算子好像写错了，应该是 depthwise_conv2d

**Comment by heavyrain-lzy** [comment]
> 认领 34, 35, 36
> 
> 36 的算子好像写错了，应该是 depthwise_conv2d

已修改，谢谢。

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
16 | shape | 中等 | @GreatV |#54221

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
13 | matmul | 困难| @GreatV | #54338

**Comment by RedContritio** [comment]
认领 4, 5

**Comment by gouzil** [comment]
> 认领 4, 5

红师傅等等
#53816

**Comment by RedContritio** [comment]
> > 认领 4, 5
> 
> 红师傅等等 #53816

感谢提醒首先 💖

然后，这个已经解耦合了，develop 中，没有这两个算子被外部引用的情况。🔅

**Comment by gouzil** [comment]
> 
> 感谢提醒首先 💖
> 
> 然后，这个已经解耦合了，develop 中，没有这两个算子被外部引用的情况。🔅

其实我想说的是https://github.com/PaddlePaddle/Paddle/pull/53816#issuecomment-1578112545 这个，不过你想试试也行

**Comment by GreatV** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
10 | exponential_ | 容易| @GreatV |

**Comment by ccsuzzh** [comment]
任务序号 | 算子名 | 困难程度 | 认领人 | 相关PR
-- | -- | -- | -- | --
9 | expand | 容易 |  @ccsuzzh  | #54628

**Comment by luotao1** [comment]
算子定义生成体系建设--静态图算子自动生成-第二期 已全部完成，感谢参与的小伙伴们！
> 按 merge 的时间顺序，排名不分先后： @GreatV (8) @huangjiyi (10) @gouzil (6) @Liyulingyue (2) @enkilee (2) @longranger2 (2) @ccsuzzh (2) @sanbuphy (1)


欢迎继续参与[快乐开源的其他任务](https://github.com/PaddlePaddle/Paddle/issues/48019)！
